src/sql/workbench/api/node/extHostQueryEditor.ts

```diff
@@ -63,4 +57,11 @@ export class ExtHostQueryEditor implements ExtHostQueryEditorShape {
 			listener.onQueryEvent(event.type, new ExtHostQueryDocument('MSSQL', fileUri, this._proxy), planXml);
 		}
 	}
+
+	public $getQueryDocument(fileUri: string): Thenable<azdata.queryeditor.QueryDocument> {
+		return new Promise((resolve) => {
+			resolve(new ExtHostQueryDocument('MSSQL', fileUri, this._proxy));
```

Comment:
update here to either fetch provider from the main thread or to remove the providerId field (in which case the extension would be expected to know the provider ID for the URI already).